docs/4.0/extend/approach.md

```diff
@@ -1,5 +1,70 @@
 ---
 layout: docs
 title: Approach
+description: Learn about the guiding principles, strategies, and techniques used to build and maintain Bootstrap so you can more easily customize and extend it yourself.
+group: extend
 ---
 
+While the getting started pages provide an introductory tour of the project and what it offers, this document focuses on _why_ we do the things we do in Bootstrap. It's meant to clearly explain our philosophy to building on the web so that others can not only learn and contribute alongside us, but help us improve our own approach.
+
+See something that doesn't sound right, or perhaps could be done better? [Open an issue](https://github.com/twbs/bootstrap/issues/new)—we'd love to discuss it with you.
+
+## Summary
+
+We'll dive into each of these more throughout, but at a high level, here's what guides our approach.
+
+- Class should be responsive-up (aka, mobile-first)
+- Components should be built with a base class and extended via modifier classes
+- Component states should obey a common z-index scale: default (initial), hover (1), active (2), focus (3)
+- Whenever possible, prefer a HTML+CSS implementation over JavaScript
+- Whenever possible, use utilities over custom styles
+- Whenever possible, avoid enforcing strict HTML requirements (children selectors)
+
+## Responsive-up
+
+- sometimes called mobile-first, but that can be a misnomer
+- styles apply to a min-width and up
+
+## Base and modifier classes
+
+- Components should be built with a base class that houses common, not-to-be overridden property-value pairs.
```

Comment:
Should we add a comment about when to create a modifier class and when to just create a new component. I'd say that if we need to modify the color (like primary, warning, etc) or the scale (like lg or sm) of the component then we generate a modifier, if it's anything else then it's a new component.